java/android/src/main/java/edu/isu/capstone/bookrec/android/urlfetcher/VolleySingleton.java

```diff
@@ -0,0 +1,40 @@
+package edu.isu.capstone.bookrec.android.urlfetcher;
+
+import android.content.Context;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+
+// Reference Docs: https://developer.android.com/training/volley/requestqueue
+public class VolleySingleton {
+    private static VolleySingleton volleyInstance;
+    private RequestQueue requestQueue;
+    private static Context context;
+
+    private VolleySingleton(Context ctx) {
+        context = ctx;
+        requestQueue = getRequestQueue();
+    }
+
+    public static synchronized VolleySingleton getInstance(Context ctx) {
```

Comment:
This isn't the best method for creating lazily loaded singletons in java. Namely, this can cause slow down from the method being synchronized. Try using the [Initialization on Demand Singleton Pattern](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom) specifically for java.

Reply:
only problem with the one i mentioned is it will fail if there is an exception while creating VolleySingleton